shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/variantcallers/somatic/MuTect2.scala

```diff
@@ -0,0 +1,101 @@
+package nl.lumc.sasc.biopet.pipelines.shiva.variantcallers.somatic
+
+import nl.lumc.sasc.biopet.extensions.{Awk, Bgzip, Tabix, gatk}
+import nl.lumc.sasc.biopet.extensions.bcftools.BcftoolsReheader
+import nl.lumc.sasc.biopet.extensions.gatk.SelectVariants
+import nl.lumc.sasc.biopet.extensions.igvtools.IGVToolsIndex
+import nl.lumc.sasc.biopet.utils.IoUtils
+import nl.lumc.sasc.biopet.utils.config.Configurable
+import org.broadinstitute.gatk.queue.extensions.gatk.TaggedFile
+
+class MuTect2(val parent: Configurable) extends SomaticVariantcaller {
+
+  def name = "mutect2"
+
+  override val mergeVcfResults: Boolean = false
+
+  // currently not relevant, at the moment only one somatic variant caller exists in Biopet
+  // and results from this won't be merged together with the results from other methods
+  def defaultPrio = -1
+
+  //val buildPONFromNormals: Boolean = config("build_PON_from_normals", default = false)
+
+  val runConEst = config("run_contest", default = "false")
+
+  def biopetScript(): Unit = {
+    val samplesDir: File = new File(outputDir, "samples")
+    if (!samplesDir.exists()) samplesDir.mkdir()
+
+    var renameSamples: List[String] = List()
+    var tumorSamples: List[String] = List()
+    var intermResult: File = null
+
+    if (tnPairs.size == 1) {
+      val pair = tnPairs.head
+      renameSamples = List(s"TUMOR ${pair.tumorSample}")
+      tumorSamples = List("TUMOR")
+      intermResult = new File(samplesDir, s"${pair.tumorSample}-${pair.normalSample}.$name.vcf")
+      addJobForPair(pair, intermResult)
+
+    } else {
+      var outputPerSample: List[TaggedFile] = List()
+      for (pair <- tnPairs) {
+        val pairLabel = s"${pair.tumorSample}-${pair.normalSample}"
+        val out: File = new File(samplesDir, s"$pairLabel.$name.vcf")
+        renameSamples :+= s"TUMOR.$pairLabel ${pair.tumorSample}"
+        tumorSamples :+= s"TUMOR.$pairLabel"
+        outputPerSample :+= TaggedFile(out, pairLabel)
+        addJobForPair(pair, out)
+      }
+
+      var sIndex = outputFile.getAbsolutePath.lastIndexOf(".vcf")
```

Comment:
For understanding what's going on, you should look a bit further downwards. The names of the tumor samples will be used by SelectVariants - so that there would be a file containing only tumor samples. The info that is gathered to the list 'renameSamples' will be used as an input for 'BcftoolsReheader', for renaming samples in the vcf file. After that the sample names in the vcf file will be the original ones, those that the user has given.